### PR TITLE
feat(gengapic): generate helpers for Go 1.23 iterators

### PR DESCRIPTION
**Pull request by codyoss**
This changes starts generating build tag protected adapters for using our iterator types in Go range expressions.

Related: https://github.com/googleapis/gax-go/pull/358

### CODE REVIEW

**Comment by codyoss** [comment]
requires gax changes to be fully released and rolled out to clients

**Comment by codyoss** [comment]
The new gax libraries have been rolled out to all clients. This is now safe for review.

**Review by noahdietz — APPROVED** [review]
Looks great. Let's just make sure we don't merge this until we are ready to have it in our clients e.g. in case we need to fix something and cut a release and this would be in that release.

**Comment by codyoss** [comment]
Agreed. Let's wait until 1.23 is out, which should be in about 2-3 weeks.

File: internal/gengapic/paging.go [hunk 1]
```diff
@@ -389,3 +389,20 @@ func (g *generator) pagingIter(pt *iterType) {
 		g.imports[spec] = true
 	}
 }
+
+func (g *generator) pagingIterGo123(pt *iterType) {
+	p := g.printf
+
+	p("// All returns an iterator. If an error is returned by the iterator, the")
+	p("// iterator will stop after that iteration.")
```
Comment (quartzmo) [inline internal/gengapic/paging.go:397]: Maybe mention here how to access the error?

Comment (codyoss) [reply internal/gengapic/paging.go:397]: I decided to just generate an example of the usage. LMK if that is good enough or you still think we should say more.

Comment (quartzmo) [reply internal/gengapic/paging.go:492]: I think the example is great, it makes everything clear.

**Review by quartzmo — APPROVED** [review]

**Pull request merged**